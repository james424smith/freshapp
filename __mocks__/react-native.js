import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
