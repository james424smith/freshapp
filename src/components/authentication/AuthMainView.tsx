import React from "react";
import { KeyboardAvoidingView, Platform } from "react-native";
import { ScrollView } from "react-native-gesture-handler";
import { IOS } from "../../constants/Platforms";
import styles from "../../styles/AuthStyles";
import GradientBackground from "./GradientBackground";

type Props = {
  children: JSX.Element;
};

const AuthMainView = (props: Props) => {
  return (
    <KeyboardAvoidingView
      style={styles.signUpContainer}
      behavior={Platform.OS === IOS ? "height" : undefined}
    >
      <ScrollView
        keyboardShouldPersistTaps="handled"
        style={styles.scrollViewStyle}
        contentContainerStyle={styles.scrollViewContainerStyles}
      >
        <GradientBackground>{props.children}</GradientBackground>
      </ScrollView>
    </KeyboardAvoidingView>
  );
};

export default AuthMainView;
