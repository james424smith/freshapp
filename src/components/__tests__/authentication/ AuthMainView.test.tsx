import React from "react";
import AuthMainView from "../../authentication/AuthMainView";
import { render, cleanup } from "@testing-library/react-native";
import { View } from "react-native";
import * as RN from "react-native";

describe("AuthMainView snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui when platform is ios", () => {
    RN.Platform.OS = "ios";

    const { toJSON } = render(
      <AuthMainView>
        <View />
      </AuthMainView>
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it("should render correct ui when platform is android", () => {
    RN.Platform.OS = "android";

    const { toJSON } = render(
      <AuthMainView>
        <View />
      </AuthMainView>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
