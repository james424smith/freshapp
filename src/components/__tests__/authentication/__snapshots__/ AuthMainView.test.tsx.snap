// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthMainView snapshot test should render correct ui when platform is android 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "flex": 1,
      "width": 750,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    keyboardShouldPersistTaps="handled"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <BVLinearGradient
        colors={
          Array [
            -16147523,
            -16754011,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1.5,
          }
        }
        locations={
          Array [
            0,
            0.5,
          ]
        }
        startPoint={
          Object {
            "x": 0.1,
            "y": 0,
          }
        }
        style={
          Object {
            "backgroundColor": "#f6f6f6",
            "flex": 1,
            "width": 750,
          }
        }
      >
        <View />
      </BVLinearGradient>
    </View>
  </RCTScrollView>
</View>
`;

exports[`AuthMainView snapshot test should render correct ui when platform is ios 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#f6f6f6",
      "flex": 1,
      "width": 750,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    keyboardShouldPersistTaps="handled"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <BVLinearGradient
        colors={
          Array [
            4278819773,
            4278213285,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1.5,
          }
        }
        locations={
          Array [
            0,
            0.5,
          ]
        }
        startPoint={
          Object {
            "x": 0.1,
            "y": 0,
          }
        }
        style={
          Object {
            "backgroundColor": "#f6f6f6",
            "flex": 1,
            "width": 750,
          }
        }
      >
        <View />
      </BVLinearGradient>
    </View>
  </RCTScrollView>
</View>
`;
